Patch to fix #531522.
This patch is the cumulation of upstream patches:

    https://svn.open-mpi.org/trac/ompi/changeset/21479
    https://svn.open-mpi.org/trac/ompi/changeset/21489

This change might make it into OMPI v1.3.3.
See http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=531522#181


--- openmpi-1.3.2.orig/opal/mca/memory/ptmalloc2/hooks.c
+++ openmpi-1.3.2/opal/mca/memory/ptmalloc2/hooks.c
@@ -721,10 +721,24 @@
        (unfortunately, there's really no good way to do this other
        than this abstraction violation :-( ) */
     struct stat st;
+    check_result_t r1, r2;
     check_result_t lp = check("OMPI_MCA_mpi_leave_pinned");
     check_result_t lpp = check("OMPI_MCA_mpi_leave_pinned_pipeline");
     bool want_rcache = false, found_driver = false;
 
+    /* See if we want to disable this component.  This check is
+       necessary for some environments -- for example, Debian's
+       "fakeroot" build environment allocates memory during stat(),
+       causing Badness (see http://bugs.debian.org/531522).
+       $FAKEROOTKEY is set by Debian's "fakeroot" build environment;
+       check for that explicitly. */
+    r1 = check("OMPI_MCA_memory_ptmalloc2_disable");
+    r2 = check("FAKEROOTKEY");
+    if ((RESULT_NOT_FOUND != r1 && RESULT_NO != r1) ||
+        (RESULT_NOT_FOUND != r2 && RESULT_NO != r2)) {
+        return;
+    }
+
     /* Look for sentinel files (directories) to see if various network
        drivers are loaded (yes, I know, further abstraction
        violations...).
